python/pyarrow/tests/test_dataset.py

```diff
@@ -3621,6 +3621,204 @@ def compare_tables_ignoring_order(t1, t2):
     assert not extra_file.exists()
 
 
+def _generate_random_int_array(size=4, min=1, max=10):
+    return np.random.randint(min, max, size)
+
+
+def _generate_data_and_columns(num_of_columns, records_per_row,
+                               unique_records=None):
+    data = []
+    column_names = []
+    if unique_records is None:
+        unique_records = records_per_row
+    for i in range(num_of_columns):
+        data.append(_generate_random_int_array(size=records_per_row,
+                                               min=1,
+                                               max=unique_records))
+        column_names.append("c" + str(i))
+    return data, column_names
+
+
+def _get_num_of_files_generated(base_directory):
+    file_dirs = os.listdir(base_directory)
+    number_of_files = 0
+    for _, file_dir in enumerate(file_dirs):
```

Comment:
I used the enumerate with the idea of it being faster. But, I see your point. Just curious, if the list is too long, is it wise to use enumerate?

Reply:
`os.walk` is a nice suggestion. Updated the code.

Reply:
Yeah `glob('**/*.parquet')` will actually work for any level of nesting, so that sounds good to me. (See https://stackoverflow.com/questions/32604656/what-is-the-glob-character)

Reply:
Great!